v21.2/performance-best-practices-overview.md

```diff
@@ -323,49 +323,59 @@ If you have long-running queries (such as analytics queries that perform full ta
 
-There are two levels of contention:
+    Transaction contention can also increase the rate of transaction restarts, and thus make the proper implementation of [client-side transaction retries](transactions.html#client-side-intervention) more critical.
 
-- Transactions that operate on the same range but different index key values will be limited by the overall hardware capacity of a single node (the range lease holder).
+### Resource contention
 
-- Transactions that operate on the same index key values (specifically, that operate on the same [column family](column-families.html) for a given index key) will be more strictly serialized to obey transaction isolation semantics.
+Resource contention occurs when a range is indexed on a column of data that is sequential in nature such that all incoming writes to the range will be the last (or first) item in the index and appended to the end of the range. As a result, the system cannot find a point in the range that evenly divides the traffic, and the range cannot benefit from load-based splitting, creating a hotspot on the single range.
 
-Transaction contention can also increase the rate of transaction restarts, and thus make the proper implementation of [client-side transaction retries](transactions.html#client-side-intervention) more critical.
+To avoid resource contention, use index key values with a random distribution of values, so that transactions over different rows are more likely to operate on separate data ranges. See the [SQL FAQs](sql-faqs.html) on row IDs for suggestions.
 
-### Find contention
+<a id="understanding-and-avoiding-transaction-contention"></a>
 
-{% include {{ page.version.version }}/performance/statement-contention.md %}
+### Transaction contention
 
-### Avoid contention
+Transaction contention occurs when the following three conditions are met:
 
-To avoid contention, you can apply multiple strategies:
+- There are multiple concurrent transactions or statements (sent by multiple clients connected simultaneously to a single CockroachDB cluster).
+- They operate on the same data, specifically over table rows with the _same index key values_ (either on [primary keys](primary-key.html) or secondary [indexes](indexes.html)) or using index key values that are close to each other, and thus place the indexed data on the same [data ranges](architecture/overview.html).
+- At least one of the transactions modify the data.
 
-- Use index key values with a more random distribution of values, so that transactions over different rows are more likely to operate on separate data ranges. See the [SQL FAQs](sql-faqs.html) on row IDs for suggestions.
+A set of transactions that all contend on the same keys will be limited in performance to the maximum processing speed of a single node (limited horizontal scalability). Non-contended transactions are not affected in this way.
```

Comment:
This seems to be describing hot spots. Contending on the same keys would mainly cause transactions to wait (or restart) - it's less of a resource issue. @ajwerner can you confirm?